src/charm.py

```diff
@@ -530,16 +528,15 @@ def _look_up_and_set_ingress_class(self, api, body):
                 return
 
             if len(defaults) > 1:
+                default_ingress = " ".join(sorted(defaults))
                 LOGGER.warning(
                     "Multiple default ingress classes defined, declining to choose between them. "
-                    "They are: {}".format(" ".join(sorted(defaults)))
+                    f"They are: {default_ingress}"
                 )
                 return
 
             ingress_class = defaults[0]
-            LOGGER.info(
-                "Using ingress class {} as it is the cluster's default".format(ingress_class)
-            )
+            LOGGER.info(f"Using ingress class {ingress_class} as it is the cluster's default")
```

Comment:
Thank you for updating the code base to use f-strings, there is an exception to using f-strings in log messages as it is a security risk: https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#formatting-log-messages

Reply:
Done :)